tests/integration/allocator_accessibility.cpp

```diff
@@ -0,0 +1,175 @@
+{
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe"; 
+  if(is_accessible(cPlatform::host, *m_allocator)) {
+    double* data = static_cast<double*>(m_allocator->allocate(m_size * sizeof(double)));
+    ASSERT_NO_THROW(data[0] = m_size*m_size);
+  } else {
+    double* data = static_cast<double*>(m_allocator->allocate(m_size * sizeof(double)));
+    ASSERT_DEATH(data[0] = m_size*m_size, "");
+  }
+}
+
+TEST_P(AllocatorAccessibilityTest, AccessibilityFromUndefined)
+{
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe"; 
+  if(is_accessible(cPlatform::undefined, *m_allocator)) {
+    FAIL() << "An Undefined platform is not accessible." << std::endl;
+  } else {
+    SUCCEED(); //Succeed every time we can't access an undefined platform.
+  }
+}
+
+#if defined(UMPIRE_ENABLE_CUDA)
+TEST_P(AllocatorAccessibilityTest, AccessibilityFromCuda)
+{
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe"; 
+  if(is_accessible(cPlatform::cuda, *m_allocator)) {
+    double* data = static_cast<double*>(m_allocator->allocate(m_size * sizeof(double)));
+    ASSERT_NO_THROW(test(data, m_size));
+  } else {
+    if(m_allocator->getAllocationStrategy()->getTraits().resource == myResource::FILE)
+      SUCCEED(); //FILE should not be accessed from CUDA
+    else {
+      double* data = static_cast<double*>(m_allocator->allocate(m_size * sizeof(double)));
+      ASSERT_DEATH(test(data, m_size), "");
+    }
+  }
+}
+#endif
+
+#if defined(UMPIRE_ENABLE_HIP)
+TEST_P(AllocatorAccessibilityTest, AccessibilityFromHip)
+{
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe"; 
+  if(is_accessible(cPlatform::hip, *m_allocator)) {
+    double* data = static_cast<double*>(m_allocator->allocate(m_size * sizeof(double)));
+    ASSERT_NO_THROW(test(data, m_size));
+  } else {
+    if(m_allocator->getAllocationStrategy()->getTraits().resource == myResource::FILE)
+      SUCCEED(); //FILE should not be accessed from HIP
+    else {
+      double* data = static_cast<double*>(m_allocator->allocate(m_size * sizeof(double)));
+      ASSERT_DEATH(test(data, m_size), "");
+    }
+  }
+}
+#endif
+
+#if defined(UMPIRE_ENABLE_OPENMP_TARGET)
+TEST_P(AllocatorAccessibilityTest, AccessibilityFromOpenMP)
+{
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe"; 
+  int dev = m_allocator->getAllocationStrategy()->getTraits().id;
+  if(is_accessible(cPlatform::omp_target, *m_allocator)) {
+    double* data = static_cast<double*>(m_allocator->allocate(m_size * sizeof(double)));
+    ASSERT_NO_THROW(test(data, m_size, dev));
+  } else {
+    if(m_allocator->getAllocationStrategy()->getTraits().resource == myResource::FILE)
+      SUCCEED(); //FILE should not be accessed from OpenMP
+    else {
+      double* data = static_cast<double*>(m_allocator->allocate(m_size * sizeof(double)));
+      ASSERT_DEATH(test(data, m_size, dev), "");
+    }
+  }
+}
+#endif
+
+/////////////////////////////
+//Sycl test not yet available
```

Comment:
This could be left as is, or I could mimic the OpenMP Target test just so that something is here. For now, there's just this placeholder.